api/src/main/java/io/strimzi/api/kafka/model/kafka/cruisecontrol/ApiUsers.java

```diff
@@ -0,0 +1,165 @@
+        this.valueFrom = valueFrom;
+    }
+
+    public Map<String, Object> getAdditionalProperties() {
+        return this.additionalProperties;
+    }
+
+    public void setAdditionalProperty(String name, Object value) {
+        this.additionalProperties.put(name, value);
+    }
+
+    /**
+     * Factory method to create and return an API Users object of the given type
+     *
+     * @return API Users object of given type
+     */
+    public static ApiUsers create(CruiseControlSpec ccSpec) {
+        if (ccSpec.getApiUsers() == null || ccSpec.getApiUsers().getType() == null) {
+            // default configuration used for Strimzi-managed API users
+            return new HashLoginServiceApiUsers();
+        }
+
+        String type = ccSpec.getApiUsers().getType();
+        switch (type) {
+            case TYPE_HASH_LOGIN_SERVICE:
+                return new HashLoginServiceApiUsers();
+            default:
+                throw new IllegalArgumentException("Unknown ApiUsers type: " + type);
+        }
+    }
+
+    /**
+     * Parse String containing API credential config into map of API user entries.
+     *
+     * @param config API credential config file as a String.
+     *
+     * @return map of API credential entries containing username, password, and role.
+     */
+    public abstract Map<String, UserEntry> parseEntriesFromString(String config);
+
+    /**
+     * Returns Cruise Control API auth credentials file as a String.
+     *
+     * @param entries Map of API user entries containing API user credentials.
+     *
+     * @return Returns Cruise Control API auth credentials file as a String.
+     */
+    public abstract String generateApiAuthFileAsString(Map<String, UserEntry> entries);
+
+    /**
+     * By default Cruise Control defines three roles: VIEWER, USER and ADMIN.
+     * For more information checkout the upstream Cruise Control Wiki here:
+     * <a href="https://github.com/linkedin/cruise-control/wiki/Security#authorization">Cruise Control Security</a>
+     */
+    public enum Role {
+        /**
+         * VIEWER: has access to the most lightweight kafka_cluster_state, user_tasks, and review_board endpoints.
+         */
+        VIEWER,
+        /**
+         * USER: has access to all the GET endpoints except bootstrap and train.
+         */
+        USER,
+        /**
+         * ADMIN: has access to all endpoints.
+         */
+        ADMIN;
+
+        public static Role fromString(String s) {
+            switch (s) {
+                case "VIEWER":
+                    return VIEWER;
+                case "USER":
+                    return USER;
+                case "ADMIN":
+                    return ADMIN;
+                default:
+                    throw new IllegalArgumentException("Unknown role: " + s);
+            }
+        }
+    }
+
+    /**
+     * Represents a single API user entry including name, password, and role.
+     */
+    public static final class UserEntry {
```

Comment:
I was wondering if this should be part of the api so exposed to the users who uses our strimzi api module. I guess this is just an internal representation for the users coming from the Secret but should not be available outside. @scholzj wdyt?

Reply:
Agree, this should not be part of the `api` module.

Reply:
I think this applies to the Enum as well as to many of the methods there. I'm quite shure that was not there when I approved it :-(.

Reply:
the only thing that should be in the `api` classes are the getters and setters for the CRD fields.